packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,2152 @@
+import * as appscaling from '../../aws-applicationautoscaling';
+import * as cloudwatch from '../../aws-cloudwatch';
+import * as iam from '../../aws-iam';
+import * as kinesis from '../../aws-kinesis';
+import * as kms from '../../aws-kms';
+import {
+  ArnFormat,
+  Aws, CfnCondition, CfnCustomResource, CfnResource, CustomResource, Duration,
+  Fn, IResource, Lazy, Names, RemovalPolicy, Resource, Stack, Token,
+} from '../../core';
+import { Construct } from 'constructs';
+import { DynamoDBMetrics } from './dynamodb-canned-metrics.generated';
+import { CfnGlobalTable, CfnTable, CfnTableProps } from './dynamodb.generated';
+import * as perms from './perms';
+import { ReplicaProvider } from './replica-provider';
+import { EnableScalingProps, IScalableTableAttribute } from './scalable-attribute-api';
+import { ScalableTableAttribute } from './scalable-table-attribute';
+
+const HASH_KEY_TYPE = 'HASH';
+const RANGE_KEY_TYPE = 'RANGE';
+
+
+// https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
+const MAX_LOCAL_SECONDARY_INDEX_COUNT = 5;
+
+/**
+ * Options for configuring a system errors metric that considers multiple operations.
+ */
+export interface SystemErrorsForOperationsMetricOptions extends cloudwatch.MetricOptions {
+
+    /**
+     * The operations to apply the metric to.
+     *
+     * @default - All operations available by DynamoDB tables will be considered.
+     */
+    readonly operations?: Operation[];
+
+}
```

Comment:
I appreciate that you copy/pasted all of this out to get started.

Now that we're working on landing this as "official" code in the CDK repository, it makes sense to start cleanup.

For all the definitions that are shared between `table.ts` and `global-table.ts`, it makes sense to move them out to a separate file (call it `shared.ts` for example), and start referencing them from there.